hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictionOperator.java

```diff
@@ -113,6 +109,30 @@ public void removeEvictableRecords(RecordStore recordStore, int evictableSize, M
         }
     }
 
+    protected boolean tryEvict(Record record, RecordStore recordStore, boolean backup, long now) {
+        String mapName = recordStore.getName();
+        Data key = record.getKey();
+        Object value = record.getValue();
+
+        if (!evictIfNotLocked(key, recordStore, backup)) {
+            return false;
+        }
+
+        if (!backup) {
+            mapServiceContext.interceptAfterRemove(mapName, value);
+
+            boolean expired = recordStore.isExpired(record, now, false);
+            if (expired) {
+                // Fire EXPIRED event also because there is a possibility that
+                // evicted entry may be also an expired one.
```

Comment:
resolved. from a discussion:
```
given that we have to maintain compatibility with the old eviction listener then we need to fire eviction event for expired entries. and because the entry is expired then we have to fire the expire event too.

later on we can add another listener which will be notified for (size-based) evictions only.
```